FraihmworkHealthAndIntegrityApi.yaml

```diff
@@ -1787,6 +2020,71 @@ components:
         - sourceId
         - description
         - timeOfBlocklist
+           
+    UnavailabilitySchedule:
+      type: object
+      required:
+        - componentId
+        - schedules
+      properties:
+        componentId:
+          type: string
+          format: uuid
+          description: |
+            The unique identifier for the component. 
+            - This can represent one or more component IDs when the schedule is applied to multiple components.
+            - In cases where multiple component IDs are used, each component's unavailability schedule should be represented as a separate object within an array.
+        schedules:
+          type: array
+          items:
+            type: object
+            required:
+              - days
+              - timePeriods
+            properties:
+              days:
+                type: array
+                items:
+                  type: string
+                  enum:
+                    - Monday
+                    - Tuesday
+                    - Wednesday
+                    - Thursday
+                    - Friday
+                    - Saturday
+                    - Sunday
+                    - All
+                description: |
+                  Specifies the days of the week when the component is unavailable.
+                  - Users can list individual days (e.g., Monday, Tuesday, All) to set specific unavailability schedules for those days.
+                  - If 'All' is selected, the component will be marked as unavailable for the entire week.
+              timePeriods:
+                type: array
+                items:
+                  type: object
+                  properties:
+                    startTime:
+                      type: string
+                      pattern: "^(2[0-3]|[01][0-9]):[0-5][0-9]$"
+                      description: |
+                        Start time of the unavailability period, formatted as "HH:MM".
+                        - If not specified and `timePeriods` is provided, defaults to the start of the day (00:00).
+                        - This allows for defining specific periods of unavailability within the day.
+                    endTime:
+                      type: string
+                      pattern: "^(2[0-3]|[01][0-9]):[0-5][0-9]$"
+                      description: |
+                        End time of the unavailability period, formatted as "HH:MM".
+                        - If not specified and `timePeriods` is provided, defaults to the end of the day (23:59).
+                        - This allows for defining specific periods of unavailability within the day.
+                  description: |
```

Comment:
This is valid as per the openapi spec, but it is kind of nice to have the description field for the object be near the start of the object declaration (instead of _after_ all the properties)

Reply:
sounds good. moved up description.